bpf/bpf_netdev.c

```diff
@@ -552,7 +567,12 @@ int from_netdev(struct __sk_buff *skb)
 			tunnel_endpoint = skb->cb[4];
 			skb->mark = 0;
 			bpf_clear_cb(skb);
-			return __encap_and_redirect_with_nodeid(skb, tunnel_endpoint, seclabel, TRACE_PAYLOAD_LEN);
+
+#ifdef ENCAP_IFINDEX
+			if (tunnel_endpoint)
+				return __encap_and_redirect_with_nodeid(skb, tunnel_endpoint, seclabel, TRACE_PAYLOAD_LEN);
```

Comment:
nit: Only this ENCAP_IFINDEX case uses {`tunnel_endpoint`, `seclabel`} so could fold those bits together under the same macro check. Probably won't make a difference to code generation, I'm guessing LLVM will detect lack of use and optimize the calls out.